app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -821,8 +821,9 @@ public void processSessionResponse(int statusCode) {
             case SessionNavigator.LAUNCH_CONFIRM_DETAIL:
                 launchConfirmDetail(asw);
                 break;
-            case SessionNavigator.EXCEPTION_THROWN:
-                displayException(sessionNavigator.getCurrentException());
```

Comment:
It did not make sense to be calling this `displayException` method for this error, since it was showing a very specific error message that was unrelated. Instead, since this is an xpath error, handle it as a user-facing error the way we do for other exceptions of this type.